mantidproject/mantid
# Add progress bar

**Description of work.**
This should be reviewed/merged after #23190

This adds a progress bar and row coloring to the table in order to track progress through a batch reduction. In order to enable this a new work handler called batch_process_runner was created.


**To test:**
In the new sans GUI try a reduction on several rows. Verify that after being reduced the rows are colored correctly and that the progress bar updates. 


Fixes #xxxx. 

This does not require release notes as only going into a feature branch


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
